lib/App/Config/Chronicle.pm

```diff
@@ -393,6 +417,397 @@ sub _add_dynamic_setting_info {
     return;
 }
 
+=head1 SUBROUTINES/METHODS
+######################################################
+###### Start new API
+######################################################
+
+=head2 local_caching
+
+If local_caching is set to the true then key-value pairs stored in Redis will be cached locally.
+
+Calling update_cache will update the local cache with any changes from Redis.
+refresh_interval defines (in seconds) the minimum time between seqequent updates.
+
+Calls to get on this object will only ever access the cache.
+Calls to set on this object will immediately update the values in the local cache and Redis.
+
+=cut
+
+has local_caching => (
+    isa     => 'Bool',
+    is      => 'ro',
+    default => 0,
+);
+
+=head2 update_cache
+
+Loads latest values from data chronicle into local cache.
+Calls to this method are rate-limited by C<refresh_interval>.
+
+=cut
+
+sub update_cache {
+    my $self = shift;
+    die 'Local caching not enabled' unless $self->local_caching;
+
+    return unless $self->_has_refresh_interval_passed();
+    $self->_updated_at(Time::HiRes::time());
+
+    return unless $self->_is_cache_stale();
+
+    my $keys = [$self->dynamic_keys(), '_global_rev'];
+    my @all_entries = $self->_retrieve_objects_from_chron($keys);
+    $self->_store_objects_in_cache({map { $keys->[$_] => $all_entries[$_] } (0 .. $#$keys)});
+
+    return 1;
+}
+
+sub _has_refresh_interval_passed {
+    my $self                   = shift;
+    my $now                    = Time::HiRes::time();
+    my $prev_update            = $self->_updated_at;
+    my $time_since_prev_update = $now - $prev_update;
+    return ($time_since_prev_update >= $self->refresh_interval);
+}
+
+sub _is_cache_stale {
+    my $self      = shift;
+    my @rev_cache = $self->_retrieve_objects_from_cache(['_global_rev']);
+    my @rev_chron = $self->_retrieve_objects_from_chron(['_global_rev']);
+    return !($rev_cache[0] && $rev_chron[0] && $rev_cache[0]->{data} eq $rev_chron[0]->{data});
+}
+
+=head2 global_revision
+
+Returns the global revision version of the config chronicle.
+This will correspond to the last time any of values were changed.
+
+=cut
+
+sub global_revision {
+    my $self = shift;
+    return ($self->get('_global_rev'));
```

Comment:
is there any special reason that return a list here ?

Reply:
I think that was an accident.